client/src/main/java/io/pravega/client/batch/impl/BatchClientImpl.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.batch.impl;
+
+import com.google.common.annotations.Beta;
+import com.google.common.collect.Iterators;
+import io.pravega.client.batch.BatchClient;
+import io.pravega.client.batch.SegmentInfo;
+import io.pravega.client.batch.SegmentIterator;
+import io.pravega.client.batch.StreamInfo;
+import io.pravega.client.netty.impl.ConnectionFactory;
+import io.pravega.client.segment.impl.Segment;
+import io.pravega.client.segment.impl.SegmentInputStreamFactory;
+import io.pravega.client.segment.impl.SegmentInputStreamFactoryImpl;
+import io.pravega.client.segment.impl.SegmentMetadataClient;
+import io.pravega.client.segment.impl.SegmentMetadataClientFactory;
+import io.pravega.client.segment.impl.SegmentMetadataClientFactoryImpl;
+import io.pravega.client.stream.Serializer;
+import io.pravega.client.stream.Stream;
+import io.pravega.client.stream.impl.Controller;
+import io.pravega.client.stream.impl.StreamCut;
+import io.pravega.client.stream.impl.StreamImpl;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import lombok.Cleanup;
+import org.apache.commons.lang3.NotImplementedException;
+
+import static io.pravega.common.concurrent.FutureHelpers.getAndHandleExceptions;
+
+@Beta
+public class BatchClientImpl implements BatchClient {
+
+    private final Controller controller;
+    private final SegmentInputStreamFactory inputStreamFactory;
+    private final SegmentMetadataClientFactory segmentMetadataClientFactory;
+
+    public BatchClientImpl(Controller controller, ConnectionFactory connectionFactory) {
+        this.controller = controller;
+        inputStreamFactory = new SegmentInputStreamFactoryImpl(controller, connectionFactory);
+        segmentMetadataClientFactory = new SegmentMetadataClientFactoryImpl(controller, connectionFactory);
+    }
+
+    @Override
+    public StreamInfo getStreamInfo(Stream stream) {
```

Comment:
Given that the implementation of this method isn't clear, should we consider having it out for now (and out of the interface)?